scipy/stats/_sobol.pyx

```diff
@@ -192,11 +215,19 @@ cdef int ibits(const cnp.uint64_t x, const int pos, const int length) nogil:
 @cython.boundscheck(False)
 @cython.wraparound(False)
 cpdef void _initialize_v(
-    cnp.uint64_t[:, ::1] v, const int dim, const int bits
+    uint_32_64[:, ::1] v, const int dim, const int bits
 ):
     """Initialize matrix of size ``dim * bits`` with direction numbers."""
     cdef int d, i, j, k, m
-    cdef cnp.uint64_t p, newv, pow2
+    cdef uint_32_64 p, newv, pow2
+    cdef uint_32_64[:] poly = get_poly_vinit(
+        'poly',
+        np.uint32 if uint_32_64 is cnp.uint32_t else np.uint64
+    )
+    cdef uint_32_64[:, ::1] vinit = get_poly_vinit(
+        'vinit',
+        np.uint32 if uint_32_64 is cnp.uint32_t else np.uint64
+    )
```

Comment:
Comment is really on the line `v[d, j] = vinit[d][j]` a little further down (but I can't post one there)

It's usually better to do `vinit[d, i]` (i.e. both indices at once). It avoids creating an intermediate memoryview.

Reply:
Good catch! A bit faster as well.